lib/actions/upload.js

```diff
@@ -0,0 +1,100 @@
+/*eslint no-invalid-this: 0 no-console: 0*/
+'use strict';
+const co = require('co');
+const Client = require('ssh2-sftp-client');
+const ip = require('../ip.js');
+const request = require('request');
+const path = require('path');
+const eioUtils = require('elasticio-node').messages;
+
+const sftp = new Client();
+
+function createConnectObject(cfg) {
+    let host = cfg.host.trim();
+
+    const protocolIndex = host.indexOf("sftp://");
+
+    if (protocolIndex === 0) {
+        host = host.substring(SFTP_PROTOCOL.length);
+    }
+
+    let port = 22;
+
+    const hostAndPort = host.split(':');
+
+    if (hostAndPort.length > 1) {
+        let portVal = hostAndPort[1];
+        host = hostAndPort[0];
+
+        if (portVal!== '' && !isNaN(portVal)) {
+            port = parseInt(portVal);
+        }
+    }
+
+    return ip.resolve(host).then(function (address) {
```

Comment:
Why not just attempt to connect here and store connection for later use? In host is not resolving `sftp.connect` will reject.